### PR TITLE
Update OpenVINO init of new GEMM layer

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

CI validation:

- [x] 2022.1.0: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/100368
- [ ] 2021.4.2: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/100373

Checklist:
- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom

build_image:Custom=ubuntu-openvino-2021.4.2:20.04
Xbuild_image:Custom=ubuntu-openvino-2022.1.0:20.04

test_modules:Custom=dnn,python2,python3,java,gapi,video

buildworker:Custom=linux-4
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Review by asmorkalov — APPROVED** [review]
:+1: 
Build with OpenVINO 2022.1.1.

**Comment by asmorkalov** [comment]
Locally failed tests:
```
[  PASSED  ] 9440 tests.
[  FAILED  ] 17 tests, listed below:
[  FAILED  ] DNNTestNetwork.FastNeuralStyle_eccv16/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] DNNTestNetwork.FastNeuralStyle_eccv16/1, where GetParam() = OCV/OCL
[  FAILED  ] Test_Model.Keypoints_face/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_Model.TextRecognitionWithCTCPrefixBeamSearch/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Linear/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Multiplication/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Constant/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.LinearWithConstant/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.MatmulWithTwoInputs/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Gemm/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Gemm_bias/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Conformance_Gemm_all_attributes/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Conformance_Gemm_alpha/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Conformance_Gemm_beta/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_layers.Conformance_Gemm_default_matrix_bias/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_nets.LResNet100E_IR/0, where GetParam() = NGRAPH/CPU
[  FAILED  ] Test_ONNX_nets.Resnet34_kinetics/0, where GetParam() = NGRAPH/CPU
```

**Comment by dkurt** [comment]
@asmorkalov, thanks, I will fix it.

**Comment by dkurt** [comment]
Remaining failed test: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/100366
```
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_ONNX_nets.Googlenet/0, where GetParam() = NGRAPH/CPU
```

File: modules/dnn/src/layers/gemm_layer.cpp [hunk 1]
```diff
@@ -289,31 +289,32 @@ class GemmLayerImpl CV_FINAL : public GemmLayer {
...
-
-            std::vector<size_t> weight_shape{(size_t)blobs[0].size[0], (size_t)blobs[0].size[1]};
-            auto ieWeights = std::make_shared<ngraph::op::Constant>(ngraph::element::f32, weight_shape, blobs[0].data);
-            matmul = std::make_shared<ngraph::op::MatMul>(inp, ieWeights, transA, transB);
         }
 
         if (have_bias && const_C) {
-            auto bias_node = std::make_shared<ngraph::op::Constant>(ngraph::element::f32,
-                                              ngraph::Shape{(size_t)blobs.back().size[1]}, blobs.back().data);
+            Mat bias = blobs.back();
```
Comment (fengyuentau) [inline modules/dnn/src/layers/gemm_layer.cpp:310]: For the GoogleNet test, I suspect a manual broadcast is needed for bias. I saw somewhere before that it seems openvino nodes cannot perform auto-broadcast.

**Comment by fengyuentau** [comment]
As for the perf tests, maybe you need to change initialization to pass the check:

https://github.com/opencv/opencv/blob/b51a78d4391429c958592cbdce1fa8426c556360/modules/dnn/src/net_openvino.cpp#L272-L282

File: modules/dnn/src/layers/gemm_layer.cpp (hunk 1 above)
Comment (dkurt) [reply modules/dnn/src/layers/gemm_layer.cpp:321]: Problem was in OpenVINO iterpretation of the batch dimension. Adding reshape from `[2, 1, 1024]` back to `[2, 1024]` solved it.

**Comment by fengyuentau** [comment]
I guess it does not make a lot of senses to run both `Gemm.gemm` and `Gemm.innerproduct` on the ngraph backend? They are both basically calling the same openvino nodes. If so, keep only one is ok to save time running tests, e.g. `Gemm.gemm`.

**Comment by dkurt** [comment]
@fengyuentau, we can keep for compatibility. There are small differences in nGraph construction related to the reshapes and others before MatMul.

**Comment by fengyuentau** [comment]
ok, makes sense then.

**Comment by asmorkalov** [comment]
the last issue on my side:
```
[ RUN      ] DNNTestNetwork.FastNeuralStyle_eccv16/0, where GetParam() = NGRAPH/CPU
/home/alexander/Projects/OpenCV/opencv-master/modules/dnn/test/test_common.impl.hpp:79: Failure
Expected: (normInf) <= (lInf), actual: 0.00276947 vs 0.002
First run  |ref| = 148.53863525390625
/home/alexander/Projects/OpenCV/opencv-master/modules/dnn/test/test_common.impl.hpp:79: Failure
Expected: (normInf) <= (lInf), actual: 0.00216675 vs 0.002
Second run  |ref| = 148.02413940429688
[  FAILED  ] DNNTestNetwork.FastNeuralStyle_eccv16/0, where GetParam() = NGRAPH/CPU (1371 ms)
```

**Comment by asmorkalov** [comment]
OpenVINO 2022.1.1 with old host (no AVX2). Not sure if the failure is related.

**Comment by dkurt** [comment]
@asmorkalov, that's right, model is from Torch frameworks and this PR is only related to ONNX models.

**Review by fengyuentau — APPROVED** [review]
👍 I guess failed tests in gapi does not relate to these changes, right?

**Comment by dkurt** [comment]
@fengyuentau , yes, the problem is old: https://github.com/opencv/opencv/issues/24049

**Comment by asmorkalov** [comment]
@dkurt Could you take a look on the BuildBot test failure:
```
[ RUN      ] Test_ONNX_layers.OpenAI_CLIP_head/0, where GetParam() = NGRAPH/CPU
[ INFO:0@88.629] global onnx_importer.cpp:828 populateNet DNN/ONNX: loading ONNX v8 model produced by 'clip-vit-base-head'. Number of nodes = 5, initializers = 5, inputs = 1, outputs = 1
[ INFO:0@88.629] global onnx_importer.cpp:729 parseOperatorSet DNN/ONNX: ONNX opset version = 18
[ INFO:0@88.629] global onnx_importer.cpp:1002 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Mul]:(onnx_node_output_0!input_mul) from domain='ai.onnx'
[ INFO:0@88.629] global onnx_importer.cpp:1002 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Expand]:(onnx_node_output_0!expand) from domain='ai.onnx'
[ INFO:0@88.629] global onnx_importer.cpp:1002 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Concat]:(onnx_node_output_0!concat) from domain='ai.onnx'
[ INFO:0@88.629] global onnx_importer.cpp:1002 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Gather]:(onnx_node_output_0!gather) from domain='ai.onnx'
[ INFO:0@88.629] global onnx_importer.cpp:1002 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Add]:(onnx_node_output_0!output) from domain='ai.onnx'
[ INFO:0@88.629] global net_openvino.cpp:705 switchToOpenVINOBackend DNN: switching to OpenVINO backend... (networkID=16678)
[ INFO:0@88.629] global ie_ngraph.cpp:766 initPlugin DNN/IE: Calling LoadNetwork(device=CPU)...
/build/precommit_custom_linux/4.x/opencv/modules/dnn/test/test_common.impl.hpp:76: Failure
Expected: (normL1) <= (l1), actual: 2.5 vs 1e-05
clip-vit-base-head  |ref| = 6.8675580024719238
/build/precommit_custom_linux/4.x/opencv/modules/dnn/test/test_common.impl.hpp:79: Failure
Expected: (normInf) <= (lInf), actual: 6 vs 0.0001
clip-vit-base-head  |ref| = 6.8675580024719238
[  FAILED  ] Test_ONNX_layers.OpenAI_CLIP_head/0, where GetParam() = NGRAPH/CPU (5 ms)
```

**Comment by fengyuentau** [comment]
> Test_ONNX_layers.OpenAI_CLIP_head/0

Reference outputs of clip head is fixed via https://github.com/opencv/opencv/pull/24295. You can rebase opencv_extra and see whether it fixes the problem.

**Comment by fengyuentau** [comment]
DNN crash is being fixed in https://github.com/opencv/opencv/pull/24315

**Pull request merged**